index.bs

```diff
@@ -461,9 +462,14 @@ run these steps:
   3. Let |result| be a newly constructed SanitizerOptions dictionary.
   4. For any non-empty member of |config| whose key is declared in
      SanitizerOptions, copy the value to |result|.
-  5. Return |result|.
+  5. All element names in |result| should be converted to lower case.
```

Comment:
I don't think we should convert when getting the config, but rather when accepting the config during construction.

Reply:
Hmm. I don't think this makes an observable difference. But changes anyways...

Related: I was trying to figure out whether this would save me from specifiying comparisons as case-independent. What I find in DOM is that [tagName is always upper-cased](https://dom.spec.whatwg.org/#dom-element-tagname) and is arguably the wrong thing anyhow because we should compare to localName, and localName doesn't seem to have any case guarantees. With that, I do wonder why we even want force-lowercase in the first place.

Reply:
hm, this is funny. I thought I had a reference where localname is always lower-cased. But now I can't find it anymore... 
@hsivonen brought this up in code review. Maybe he remembers?

Reply:
Note to self: I had assumed tagName and local name were the same, which they aren't. tagName is a legacy concept and uppercase. local name is the current concept and (mostly) lowercase.

The closest definition I can find is: https://dom.spec.whatwg.org/#dom-document-createelement

This says that local names are always lower-cased in HTML documents (but not XML ones). I'm going to assume that we'll forfit XML docs and just always lowercase. I'll change the PR to consistently refer to local names, and not to tag names. (Also the spec, in case I'll find additional references there.)

Reply:
Might there be an issue if the sanitizer were to support [SVG elements](https://developer.mozilla.org/en-US/docs/Web/SVG/Element), many of which have mixed-case `localName` values?

For example,

```javascript
document.createElementNS("http://www.w3.org/2000/svg", "linearGradient").localName; // "linearGradient"
```

In my testing with the Chrome 91 browser, the `localName` is mixed-case even when the SVG element are embedded in an HTML document.

```javascript
document.querySelector("linearGradient").localName; // "linearGradient"
```

Reply:
Sorry I'm a bit late, I just wanted to confirm that this is indeed a problem for SVG & MathML elements.

Right now, we're cheating by simply ignoring non-HTML. But clearly SVG & MathML are on the agenda. So we need to make sure to not inadvertently block our options there.

Reply:
The HTML tokenizer always lower-cases incoming element names. For SVG, it then [re-adjusts the case for a list of elements](https://html.spec.whatwg.org/#parsing-main-inforeign:svg-namespace).

Does the Sanitizer API consider allow-listing elements that are in a namespace other than HTML, SVG, and MathML to be in scope for the API?

Reply:
> Does the Sanitizer API consider allow-listing elements that are in a namespace other than HTML, SVG, and MathML to be in scope for the API?

Strong "No!" :)

Reply:
Thanks all for the feedback. Very extra thanks fo @hsivonen for the spec reference, because I think that provides us with a good way forward:

Instead of prescribing either lower or uppercase, we'll spec it as normalize-in-whatever-fashion-HTML-does. That should match the principle of least surprise, and also allow for easy implementation, since presumably all user agents have the suitable normalization already implemented somewhere.